### PR TITLE
[sfputil] Display 'N/A' for non-SFP ports

### PR DESCRIPTION
**Pull request by PJHsieh**
Currently, "Not Present" will be shown at non-sfp ports.

This enhancement will show "N/A" instead of "Not Present"

at non-sfp ports.

**- What I did**
Let "N/A" be show at non-sfp ports.

**- How I did it**
Shows "Not Present" at sfpuitl valid ports.
Shows "N/A" at sfputil invalid ports.

**- How to verify it**
Execute sudo sfputil show presence at a copper, and "N/A" is output.

**- Previous command output (if the output of a command-line utility has changed)**
"N/A" instead of "Not Present" will be shown at non-sfp ports.

**- New command output (if the output of a command-line utility has changed)**
None

### CODE REVIEW

File: sfputil/main.py [hunk 1]
```diff
@@ -415,7 +415,10 @@ def presence(port):
             if presence:
                 output_table.append([port_name, "Present"])
             else:
-                output_table.append([port_name, "Not present"])
+                if not platform_sfputil._is_valid_port(physical_port):
+                    output_table.append([port_name, "N/A"])
+                else:
+                    output_table.append([port_name, "Not present"])
```
Comment (jleveque) [inline sfputil/main.py:421]: Maybe it makes more sense written as follows?

```python
            if platform_sfputil._is_valid_port(physical_port):
                status_string = "Present" if presence else "Not present"
                output_table.append([port_name, status_string])
            else:
                output_table.append([port_name, "N/A"])
```

Comment (PJHsieh) [reply sfputil/main.py]: Hi, jleveque,

Your comment is applied.

**Comment by PJHsieh** [comment]
Hi, jleveque,
The file permissions be changed back to 0644.

**Comment by PJHsieh** [comment]
@keboliu, please review this PR, thanks.

**Comment by keboliu** [comment]
you mentioned that a "copper" cable will be judged as invalid port by "platform_sfputil._is_valid_port()", could you elaborate on the invalid cable type here? IMO copper passive cable is valid as AOC cable and can be handled by sfp utilities w/o any error?

**Comment by PJHsieh** [comment]
> you mentioned that a "copper" cable will be judged as invalid port by "platform_sfputil._is_valid_port()", could you elaborate on the invalid cable type here? IMO copper passive cable is valid as AOC cable and can be handled by sfp utilities w/o any error?

Actually, on SFP/SFP+/SFP28/QSFP/QSFPD-D are valid.

**Comment by PJHsieh** [comment]
> you mentioned that a "copper" cable will be judged as invalid port by "platform_sfputil._is_valid_port()", could you elaborate on the invalid cable type here? IMO copper passive cable is valid as AOC cable and can be handled by sfp utilities w/o any error?

Actually, this is not related to cable, but related to the cage on the switch.

**Comment by jleveque** [comment]
> Valid: One of SFP/SFP+/SFP28/QSFP/QSFP-DD can be inserted in the cage of a port.
> Invalid: None of SFP/SFP+/SFP28/QSFP/QSF-DD can be inserted in the cage of a port.

@PJHsieh: I don't understand this. Can you please elaborate?

**Comment by PJHsieh** [comment]
> > Valid: One of SFP/SFP+/SFP28/QSFP/QSFP-DD can be inserted in the cage of a port.
> > Invalid: None of SFP/SFP+/SFP28/QSFP/QSF-DD can be inserted in the cage of a port.
> 
> @PJHsieh: I don't understand this. Can you please elaborate?

Device_Example

**Comment by PJHsieh** [comment]
> > > Valid: One of SFP/SFP+/SFP28/QSFP/QSFP-DD can be inserted in the cage of a port.
> > > Invalid: None of SFP/SFP+/SFP28/QSFP/QSF-DD can be inserted in the cage of a port.
> > 
> > 
> > @PJHsieh: I don't understand this. Can you please elaborate?
> 
> Device_Example

@jleveque @keboliu , Please refer to the above figure.
Only RJ-45 cable can be inserted in  Port 1~Port 48, therefore, invalid for sfputil --- sfputil show presence output N/A
SFP/SFP+/SFP28 can be inserted in Port 49~Port 52, therefore, valid for sfputil --- sfputil show presence output "Not Present" or "Not Present"
QSFP can be inserted in Port 53~Port54, therefore, valid for sfputil --- sfputil show presence output "Not Present" or "Not Present".

**Comment by PJHsieh** [comment]
@jleveque @keboliu , do you still have any concern ?

**Comment by keboliu** [comment]
> @jleveque @keboliu , do you still have any concern ?

this is a very rare case, I mean haven't seen any platform still supporting RJ45 ports in the SONiC community before, but yes I think this issue is valid.  But I think you also need to override _is_valid_port() somewhere?

**Comment by PJHsieh** [comment]
> > @jleveque @keboliu , do you still have any concern ?
> 
> this is a very rare case, I mean haven't seen any platform supporting RJ45 ports in the SONiC community before, but yes I think this issue is valid. But I think you also need to override _is_valid_port() somewhere?

The "_is_valid_port" is defined in class class SfpUtilBase in sonic_platform_base/sonic_sfp/sfputilbase.py, and can be override by the derived class. The device vendors take the responsibility for implementing  their own _is_valid_port().

The main point for this PR is:
If this port is not valid for sfp, the 'N/A' is suggest to be shown.
If this port is valid for sfp, "Not Present" or "Present" shall be shown.

**Review by keboliu — APPROVED** [review]

**Review by jleveque — APPROVED** [review]

**Pull request merged**